libqtile/backend/x11/window.py

```diff
@@ -2149,8 +2149,15 @@ def handle_ClientMessage(self, event):  # noqa: N802
                 focus_behavior = self.qtile.config.focus_on_window_activation
                 if focus_behavior == "focus":
                     logger.debug("Focusing window")
-                    self.qtile.current_screen.set_group(self.group)
-                    self.group.focus(self)
+                    # Windows belonging to a scratchpad need to be toggled properly
+                    if isinstance(self.group, ScratchPad):
+                        for dropdown in self.group.dropdowns.values():
+                            if dropdown.window == self:
```

Comment:
```suggestion
                            if dropdown.window is self:
```
This would confirm that the window is exactly the same object.